packages/mikro-orm/src/index.ts

```diff
@@ -0,0 +1,210 @@
+import type { Options } from "@mikro-orm/core"
+import { MikroORM, wrap } from "@mikro-orm/core"
+import * as defaultEntities from "./entities"
+
+import type { Adapter } from "next-auth/adapters"
+import { isPromise } from "util/types"
+
+export * as defaultEntities from "./entities"
+
+/**
+ * The MikroORM adapter accepts a MikroORM instance or configuration and returns a NextAuth adapter.
+ * @param ormConnection can either be an instance promise or a MikroORM connection configuration (https://mikro-orm.io/docs/next/configuration#driver)
+ * @param options entities in the options object will be passed to the MikroORM init function as entities
+ * @returns
+ */
+export function MikroOrmAdapter<
+  TUser extends typeof defaultEntities.User,
```

Comment:
I would still prefer not to expose any generics. Isn't there like a general `Entity` type that would make TS happy on the `entities` option? I don't want the user to have to configure types on the adapter. It doesn't mean that much to me if its not perfect internally, tests should be added to cover custom entities instead.

Reply:
I think as its now, it is unnecessary anyway. Will try and drop it.